### ISSUE TITLE
Calling `getShippingMethod()` on a cart returns nothing for Postie-provided shipping methods

### ISSUE THREAD

**Issue by engram-design** [craftcms/commerce#2279]
In Commerce 3.4.x, calling `{{ cart.shippingMethod.name }}` now returns `null` during checkout, despite having the shipping method applied on the order, which can be seen reflected in the control panel for the active order. This is happening for Postie-provided shipping methods, which are provided on-the-fly.

This is due to refactored code for [`Order::getShippingMethod()`](https://github.com/craftcms/commerce/blob/develop/src/elements/Order.php#L3008) which calls `getShippingMethodByHandle()` which in turn calls `getAllShippingMethods()`.

My first issue is that the docblock for the `getAllShippingMethods()` method states:

```
Returns the Commerce managed and 3rd party shipping methods
```

Which doesn't seem true at all, and even in previous Commerce versions, this has never been the case. Here, it should return both "normal" shipping methods defined by the user in the control panel for Commerce's setting, and any shipping methods provided by plugins or modules. It has only ever done the former?

Secondly, the `Order::getShippingMethod()` has drastically changed from 3.3.5.1

https://github.com/craftcms/commerce/blob/66bb55105df6838a9967f3cc70c4b3ca0fcf0024/src/elements/Order.php#L2976-L3012

Most importantly, the previous function was using `Plugin::getInstance()->getShippingMethods()->getAvailableShippingMethods()` which was returning the correct collection because it actually includes the calls to `EVENT_REGISTER_AVAILABLE_SHIPPING_METHODS` which plugins and modules use.

So I suppose one of these sides needs fixing. Either the Order `getShippingMethod()` should call `getAvailableShippingMethods()` or the `getAllShippingMethods()` service method needs to include shipping methods provided by the event. Or - something else I'm missing.

Also refer to issue https://github.com/verbb/postie/issues/72 which is instance of the issue.

**Additional info**
- Craft version: 3.7.4
- Commerce version: 3.4.0.2
- PHP version: 7.2
- Other Plugins & versions: Postie 2.4.8

**Comment by pierrestoffe** [comment]
I'm encountering the same issue with shipping methods that are defined in a plugin via the `EVENT_REGISTER_AVAILABLE_SHIPPING_METHODS` event.

**Comment by lukeholder** [comment]
This should now be fixed in 3.4.0.3

I installed Postie, configured Fastway, and can confirm that I can select a Fastway shipping option and have the price applied to the order:


https://user-images.githubusercontent.com/133571/126593645-4e627c5d-bb9f-44d3-9567-8d43e2dd7f60.mp4

**Comment by engram-design** [comment]
@lukeholder Can confirm this works, thanks!
